### PR TITLE
feat: add "xs" size to Input and all related components

### PR DESCRIPTION
**Pull request by dodas**
## 📝 Description

Similar to the `Button` component, I am adding support for `xs` size to `Input` and all related components, such as  `Select`, `PinInput`, `Textarea` or `NumberInput`.  I also updated the Docs to fix mistakes and be more consistent.

## 💣 Is this a breaking change (Yes/No):

No.

### CODE REVIEW

File: packages/input/tests/__snapshots__/input.test.tsx.snap [hunk 1]
```diff
@@ -16,8 +16,6 @@ exports[`Elements inside input render correctly 1`] = `
   width: 2.5rem;
   height: 2.5rem;
   font-size: 1rem;
-  padding-left: 1rem;
```
Comment (dodas) [inline packages/input/tests/__snapshots__/input.test.tsx.snap:19]: I am not quite sure about these changes.
I just changed the code from using `pl` and `pr` to `px`.

File: packages/number-input/tests/__snapshots__/number-input.test.tsx.snap [hunk 2]
```diff
@@ -27,6 +27,7 @@ exports[`should render correctly 1`] = `
   padding-right: 1rem;
   height: 2.5rem;
   border-radius: 0.375rem;
+  vertical-align: top;
```
Comment (dodas) [inline packages/number-input/tests/__snapshots__/number-input.test.tsx.snap:30]: When using `NumberInput` with `size="xs"`, the stepper is slightly shifted down. This fixes it.

File: packages/theme/src/components/number-input.ts [hunk 3]
```diff
@@ -32,19 +33,25 @@ const baseStyle = (props: Record<string, any>) => ({
...
+    xs: "sm",
   }
 
+  const resolvedFontSize = Typography.fontSizes[sizeStyle.field.fontSize]
+
   return {
-    field: sizeStyle?.field,
+    field: {
+      ...sizeStyle.field,
+      verticalAlign: "top",
```
Comment (dodas) [inline packages/theme/src/components/number-input.ts:51]: When using NumberInput with size="xs", the stepper is slightly shifted down.
This fixes it.

File: packages/theme/src/components/number-input.ts [hunk 4]
```diff
@@ -32,19 +33,25 @@ const baseStyle = (props: Record<string, any>) => ({
...
+  const sizeStyle = Input.sizes[size]
 
   const radius = {
     lg: "md",
     md: "md",
     sm: "sm",
+    xs: "sm",
   }
 
+  const resolvedFontSize = Typography.fontSizes[sizeStyle.field.fontSize]
```
Comment (dodas) [inline packages/theme/src/components/number-input.ts:46]: Is this fine to do? I am trying to get rid of that `size === "lg" ? "14px" : "10px"` in stepper fontSize.

File: packages/theme/src/components/number-input.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import { mode } from "@chakra-ui/theme-tools"
 import Input from "./input"
-import Typography from "../foundations/typography"
+import typography from "../foundations/typography"
```
Comment (dodas) [inline packages/theme/src/components/number-input.ts:3]: This is really weird lint rule :)

**Review by segunadebayo — COMMENTED** [review]
Good work here @dodas. Reviewing the UI in Storybook and will leave a comment or merge shortly.

**Comment by dodas** [comment]
Thanks!

**Pull request merged**